src/mahoji/lib/abstracted_commands/buyingTripCommand.ts

```diff
@@ -0,0 +1,78 @@
+import type { ChatInputCommandInteraction } from 'discord.js';
+import { Time } from 'e';
+import { Bank } from 'oldschooljs';
+
+import { formatDuration } from '@oldschoolgg/toolkit/util';
+import type { Buyable } from '../../../lib/data/buyables/buyables';
+import type { BuyActivityTaskOptions } from '../../../lib/types/minions';
+import { itemNameFromID } from '../../../lib/util';
+import addSubTaskToActivityTask from '../../../lib/util/addSubTaskToActivityTask';
+import { calcMaxTripLength } from '../../../lib/util/calcMaxTripLength';
+import calculateShopBuyCost from '../../../lib/util/calculateShopBuyCost';
+import getOSItem from '../../../lib/util/getOSItem';
+import { handleMahojiConfirmation } from '../../../lib/util/handleMahojiConfirmation';
+import { updateBankSetting } from '../../../lib/util/updateBankSetting';
+
+export async function buyingTripCommand(
+	user: MUser,
+	channelID: string,
+	buyable: Buyable,
+	quantity: number | null,
+	interaction: ChatInputCommandInteraction
+) {
+	let quantityPerHour = buyable.quantityPerHour!;
+	if (buyable.name === 'Coal') {
+		if (user.owns('Coal bag')) {
+			quantityPerHour *= 1.6;
+		}
+	}
+	const timePerItem = Time.Hour / quantityPerHour;
+	const osItem = getOSItem(buyable.name);
```

Comment:
Optional, but recommended for future: use `Items.getOrThrow` instead of `getOSItem`. If we didn't expect it to ALWAYS return a item, it would be `Items.getItem`, in this case it would be a bug if nothing was returned, so we use getOrThrow.